tests/test_device_validator.py

```diff
@@ -266,3 +268,49 @@ def test_finite_diesel_generator(self):
         self.assertIsNone(validate_finite_diesel_generator(energy_rate=100))
         with self.assertRaises(D3ADeviceException):
             validate_finite_diesel_generator(energy_rate=-1)
+
+
+class TestHomeMeterValidator(unittest.TestCase):
+    """Tests for the HomeMeterValidator class."""
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        """Set up requirements for all individual tests."""
+        cls.validator = HomeMeterValidator
+
+    @patch.object(HomeMeterValidator, "validate_price")
+    @patch.object(HomeMeterValidator, "validate_energy")
+    def test_validate(self, validate_energy_mock, validate_price_mock):
+        """The validate method correctly calls the individual validation methods."""
+        self.validator.validate()
+        validate_price_mock.assert_called_once_with()
+        validate_energy_mock.assert_called_once_with()
+
+    @patch("d3a_interface.device_validator.validate_range_limit")
+    def test_validate_price_executes(self, validate_range_limit_mock):
+        """The validation is executed when suitable arguments are provided.
+
+        Note: this does not necessarily mean that the validation is successful, as it also depends
+        on the validate_range_limit method (which we are mocking here).
+        """
+        arguments = {
+            "initial_buying_rate": 10, "final_buying_rate": 13,
+            "energy_rate_increase_per_update": 1,
+            "initial_selling_rate": 13, "final_selling_rate": 10,
+            "energy_rate_decrease_per_update": 2
+        }
+        self.validator.validate_price(**arguments)
+        assert validate_range_limit_mock.call_count == 6
+
+    def test_validate_price_fails(self):
+        """The validation fails when incompatible arguments are provided."""
+        failing_arguments = [
+            {"initial_buying_rate": 15, "final_buying_rate": 13},
+            {"energy_rate_increase_per_update": 1, "fit_to_limit": True},
+            {"initial_selling_rate": 10, "final_selling_rate": 13},
+            {"energy_rate_decrease_per_update": 1, "fit_to_limit": True}
+        ]
+        for arguments in failing_arguments:
+            with self.subTest(arguments=arguments):
+                with self.assertRaises(D3ADeviceException):
```

Comment:
I'm not convinced why should we use subTest here? 
IMO the test is mature enough without it, no need for individual context for each of these assertions and the test should fail as soon as any validation passed without raising `D3ADeviceException`

Reply:
We have several scenarios in which the validation should fail. You can think of each of these cases as a separate method (and they should be, in my opinion, but that's out of scope). 
Using just one set of failing arguments would not cover all cases. Suppose I mistakenly change the implementation of one of the `if` clauses in the `validate_price()` method, I will want tests to fail.